src/FlipMove.js

```diff
@@ -575,38 +552,99 @@ class FlipMove extends Component {
           // because they don't have instances.
           if (!element) {
             return;
           }
 
-          const domNode = getNativeNode(element);
+          const domNode: ?HTMLElement = getNativeNode(element);
 
           // If this is the first render, we need to create the data entry
-          if (!this.childrenData[child.key]) {
-            this.childrenData[child.key] = {};
+          const key = getKey(child);
+          if (!this.childrenData[key]) {
+            this.childrenData[key] = {};
           }
 
-          this.childrenData[child.key].domNode = domNode;
+          this.childrenData[key].domNode = domNode;
         },
       })
     ));
   }
 
+  componentDidMount() {
+    // Run our `appearAnimation` if it was requested, right after the
+    // component mounts.
+    const shouldTriggerFLIP = (
+      this.props.appearAnimation &&
+      !this.isAnimationDisabled(this.props)
+    );
+
+    if (shouldTriggerFLIP) {
+      this.prepForAnimation();
+      this.runAnimation();
+    }
+  }
+
+  componentWillReceiveProps(nextProps: ConvertedProps) {
+    // When the component is handed new props, we need to figure out the
+    // "resting" position of all currently-rendered DOM nodes.
+    // We store that data in this.parent and this.children,
+    // so it can be used later to work out the animation.
+    this.updateBoundingBoxCaches();
+
+    // Convert opaque children object to array.
+    const nextChildren: Array<Element<*>> = Children.toArray(nextProps.children);
+
+    // Next, we need to update our state, so that it contains our new set of
+    // children. If animation is disabled or unsupported, this is easy;
+    // we just copy our props into state.
+    // Assuming that we can animate, though, we have to do some work.
+    // Essentially, we want to keep just-deleted nodes in the DOM for a bit
+    // longer, so that we can animate them away.
+    this.setState({
+      children: this.isAnimationDisabled(nextProps)
+        ? nextChildren.map(element => ({ ...element, element }))
+        : this.calculateNextSetOfChildren(nextChildren),
+    });
+  }
+
+  componentDidUpdate(previousProps: ConvertedProps) {
+    // If the children have been re-arranged, moved, or added/removed,
+    // trigger the main FLIP animation.
+    //
+    // IMPORTANT: We need to make sure that the children have actually changed.
+    // At the end of the transition, we clean up nodes that need to be removed.
+    // We DON'T want this cleanup to trigger another update.
+
+    const oldChildrenKeys: Array<string> = Children.toArray(this.props.children).map(d => d.key);
+    const nextChildrenKeys: Array<string> =
+        Children.toArray(previousProps.children).map(d => d.key);
+
+    const shouldTriggerFLIP = (
+      !arraysEqual(oldChildrenKeys, nextChildrenKeys) &&
+      !this.isAnimationDisabled(this.props)
+    );
+
+    if (shouldTriggerFLIP) {
+      this.prepForAnimation();
+      this.runAnimation();
+    }
+  }
```

Comment:
Ah, sorry, I was unclear. I meant for the lifecycle methods to still be above the other custom methods, but below the initialized data (the stuff that used to be in the constructor). So they should sit just above `calculateNextSetOfChildren`.